### PR TITLE
chore: Replace `keyCode` usages with `key` for KeyboardEvents

### PR DESCRIPTION
**Pull request by diegonvs**
Some keys can be different depending the browser which is being running the application but React normalize it when creating the synthetic event on: https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/packages/react-dom/src/events/getEventKey.js#L12-L29

### How I tested It:
Regarding
> but React normalize it when creating the synthetic event

From https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values: 
>[1] Older browsers may return "Spacebar" instead of " " for the Space Bar key. Firefox did so until version 37, as did Internet Explorer 9, 10, and 11.

Spacebar: 

https://github.com/liferay/clay/pull/3513/files#diff-18ecf9fd4dc1fbbb2a2bff5e734cc990L52

1. Open our Storybook on IE11
2. Open ClayCardWithNavigation story
3. Try navigating via tabs and select the first card with Spacebar key
4. You'll see an alert opening

Closes https://github.com/liferay/clay/issues/3430

### CODE REVIEW

**Comment by wincent** [comment]
> One small change I think would be helpful is to use the keys as constants at the top of the file and then reference those instead of the string directly.

Another possibility would be to move them into a file in `clay-shared` so that you're not repeating them in a number of places. That file would also provide you with a good place to put a comment explaining why/how the key names are normalized.

Consider:

```js
const SPACE = ' ';

if (event.key === SPACE) {
   // ...
}
```

Some well-intentioned developer comes along and says, "Hm, that might not work on IE; I'm going to change that":

```js
if (event.key === ' ' || event.key === 'Spacebar') {
  // ...
}
```

And it could sneak through code review, which means that it is relatively brittle.

On the other hand, imagine this:

```js
import {Keys} from '@clayui/shared';

if (event.key === Keys.SPACE ) {
   // ...
}
```

Developer might still think, "Hm, I wonder if that will work in IE", but when they look in `clay-shared` they'll see:

```js
/**
 * Key names for comparison with `Event.prototype.key`.
 *
 * Note that because React normalizes key names in its synthetic
 * event system, we don't have to worry about browser-specific
 * edge cases (such as "Spacebar" instead of " " in IE):
 *
 * See:
 *
 * - https://github.com/facebook/react/blob/b87aabdfe1/packages/react-dom/src/events/getEventKey.js#L12-L29
 * - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values
 */
export const Keys = {
  SPACE: ' ',
  // ...
};
```

In other words, less brittle, and people are more likely to copy the pattern because they'll see `import {Keys} from '@clayui/shared';` in various places around the repo.

Just an idea for you to consider anyway.

**Comment by bryceosterhaus** [comment]
> @bryceosterhaus 
> I would say lets just start with copy/paste for each map and then we can evaluate down the road if we want to abstract to @clayui/shared



> @wincent 
> Another possibility would be to move them into a file in `clay-shared`


Sorry for the confusion @diegonvs 😂😂😂

_btw:_ I'm fine with both copypasta and shared

**Comment by diegonvs** [comment]
Force pushing for fixing a dep version mismatch error :p

**Comment by diegonvs** [comment]
Updated cd54a8191b46b0a0760ccc6ac31fbcefe6fd51bf

**Comment by bryceosterhaus** [comment]
Hmm, not sure why the clayui.com build is failing... Fails for me locally on master as well, but the latest commit built fine
Screen Shot 2020-07-17 at 7 35 26 PM

**Comment by bryceosterhaus** [comment]
Built fine at https://github.com/liferay/clay/pull/3526. @diegonvs I just triggered a re-deploy, maybe that'll fix it 🤷‍♂️

**Comment by diegonvs** [comment]
Hey @bryceosterhaus:

Greener as amazon rainforest in 80ties:
Screen Shot 2020-07-20 at 17 41 45
